### PR TITLE
Use path join methods in fetch

### LINKED ISSUE #2649: Acceptance tests: use path.join() while making fetch requests instead of use string concatenation of template strings
### Description
When we make `fetch` requests,
 we usually build url by using string concatenation `(eg. backendUrl + '/ocs/api/v1/user')` 
or by template strings ```(eg. `${backendUrl}/ocs/api/v1/user` ```

Instead of doing this we can use path.join() method from helpers to make the url better.

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Path join is used in the urls used in the acceptance tests

## Related Issue

- Fixes #2649 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/helpers/webdavHelper.js [hunk 1]
```diff
@@ -14,9 +14,8 @@ const occHelper = require('../helpers/occHelper')
  * @param {string} server - (REMOTE/LOCAL) server to use for dav path
  */
 exports.createDavPath = function (userId, element) {
-  return backendHelper.getCurrentBackendUrl() +
-    '/remote.php/dav/files/' + userId + '/' + encodeURIComponent(element)
-    .replace(/%2F/g, '/')
+  return join(backendHelper.getCurrentBackendUrl(), '/remote.php/dav/files/', userId,
+    encodeURIComponent(element).replace(/%2F/g, '/'))
```
Comment (skshetry) [inline tests/acceptance/helpers/webdavHelper.js]: Move this outside in separate variable. 
```
    encodeURIComponent(element).replace(/%2F/g, '/'))
```

File: tests/acceptance/helpers/webdavHelper.js [hunk 2]
```diff
@@ -210,7 +209,8 @@ exports.getSkeletonFile = function (filename) {
       const headers = httpHelper.createAuthHeader('admin')
       const element = join(dir.trim(), filename)
       return fetch(
-        client.globals.backend_url + `/ocs/v2.php/apps/testing/api/v1/file?file=${encodeURIComponent(element)}&absolute=true&format=json`,
+        join(client.globals.backend_url,
+          `/ocs/v2.php/apps/testing/api/v1/file?file=${encodeURIComponent(element)}&absolute=true&format=json`),
```
Comment (skshetry) [inline tests/acceptance/helpers/webdavHelper.js]:
```suggestion
          '/ocs/v2.php/apps/testing/api/v1/', `file=${encodeURIComponent(element)}&absolute=true&format=json`),
```

Or, even better, separate the query params in a separate variable.

File: tests/acceptance/helpers/webdavHelper.js [hunk 3]
```diff
@@ -221,8 +221,8 @@ exports.getSkeletonFile = function (filename) {
 
 exports.uploadFileWithContent = function (user, content, filename) {
   const headers = httpHelper.createAuthHeader(user)
-  return fetch(client.globals.backend_url +
-    '/remote.php/webdav/' + filename, {
+  return fetch(join(client.globals.backend_url,
+    '/remote.php/webdav/', filename), {
```
Comment (skshetry) [inline tests/acceptance/helpers/webdavHelper.js]: Is this too large to separate into two lines?

File: tests/acceptance/stepDefinitions/generalContext.js [hunk 4]
```diff
@@ -134,7 +134,7 @@ const setTrustedServer = function (url) {
   const body = new URLSearchParams()
   body.append('url', userSettings.replaceInlineCode(url))
   const headers = httpHelper.createAuthHeader(client.globals.backend_admin_username)
-  const postUrl = `${backendHelper.getCurrentBackendUrl()}/ocs/v2.php/apps/testing/api/v1/trustedservers?format=json`
+  const postUrl = path.join(backendHelper.getCurrentBackendUrl(), '/ocs/v2.php/apps/testing/api/v1/trustedservers?format=json')
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/generalContext.js]: in this file, `path` is a node-module, not the helper.

File: tests/acceptance/stepDefinitions/generalContext.js [hunk 5]
```diff
@@ -178,7 +178,7 @@ After(async function (testCase) {
   const headers = httpHelper.createAuthHeader(client.globals.backend_admin_username)
   const body = new URLSearchParams()
   body.append('global', 'true')
-  await fetch(`${client.globals.backend_url}/ocs/v2.php/apps/testing/api/v1/lockprovisioning`,
+  await fetch(path.join(client.globals.backend_url, '/ocs/v2.php/apps/testing/api/v1/lockprovisioning'),
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/generalContext.js]: same here

File: tests/acceptance/stepDefinitions/notificationsContext.js [hunk 6]
```diff
@@ -4,13 +4,13 @@ const httpHelper = require('../helpers/httpHelper')
...
 const assert = require('assert')
+const { join } = require('../helpers/path')
 
 When('user {string} is sent a notification', function (user) {
   const body = new URLSearchParams()
   body.append('user', user)
 
-  return fetch(
-    client.globals.backend_url + '/ocs/v2.php/apps/testing/api/v1/notifications',
+  return fetch(join(client.globals.backend_url, '/ocs/v2.php/apps/testing/api/v1/notifications'),
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/notificationsContext.js]: Better to move the url into a separate variable

File: tests/acceptance/stepDefinitions/notificationsContext.js [hunk 7]
```diff
@@ -30,7 +30,7 @@ When('the user declines all shares displayed in the notifications on the webUI',
 
 Given('app {string} has been enabled', function (app) {
   const headers = httpHelper.createAuthHeader('admin')
-  return fetch(client.globals.backend_url + '/ocs/v2.php/cloud/apps/' + app + '?format=json', {
+  return fetch(join(client.globals.backend_url, '/ocs/v2.php/cloud/apps/', app) + '?format=json', {
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/notificationsContext.js]: Better to move url into separate variable

File: tests/acceptance/helpers/webdavHelper.js (hunk 1 above)
Comment (jasson99) [reply tests/acceptance/helpers/webdavHelper.js]: fixed

File: tests/acceptance/helpers/webdavHelper.js (hunk 2 above)
Comment (jasson99) [reply tests/acceptance/helpers/webdavHelper.js]: fixed

File: tests/acceptance/stepDefinitions/generalContext.js (hunk 4 above)
Comment (jasson99) [reply tests/acceptance/stepDefinitions/generalContext.js]: fixed

File: tests/acceptance/helpers/webdavHelper.js (hunk 3 above)
Comment (jasson99) [reply tests/acceptance/helpers/webdavHelper.js]: fixed

File: tests/acceptance/stepDefinitions/generalContext.js (hunk 5 above)
Comment (jasson99) [reply tests/acceptance/stepDefinitions/generalContext.js]: fixed

File: tests/acceptance/stepDefinitions/notificationsContext.js (hunk 7 above)
Comment (jasson99) [reply tests/acceptance/stepDefinitions/notificationsContext.js]: fixed

File: tests/acceptance/stepDefinitions/notificationsContext.js (hunk 6 above)
Comment (jasson99) [reply tests/acceptance/stepDefinitions/notificationsContext.js]: fixed

File: tests/acceptance/helpers/sharingHelper.js [hunk 8]
```diff
@@ -186,9 +192,8 @@ module.exports = {
     if (sharedWithUser === true) {
       sharedWithMeText = '&shared_with_me=true'
     }
-    const apiURL = client.globals.backend_url +
-                   '/ocs/v2.php/apps/files_sharing/api/v1/shares?format=json&state=all' +
-                   sharedWithMeText
+    const apiURL = join(client.globals.backend_url,
+      '/ocs/v2.php/apps/files_sharing/api/v1/shares?format=json&state=all') + sharedWithMeText
```
Comment (dpakach) [inline tests/acceptance/helpers/sharingHelper.js]: Better use `URLSearchParams()` here

Comment (jasson99) [reply tests/acceptance/helpers/sharingHelper.js]: fixed

File: tests/acceptance/helpers/sharingHelper.js [hunk 9]
```diff
@@ -182,13 +188,15 @@ module.exports = {
...
+      params.set('shared_with_me', 'true')
     }
-    const apiURL = client.globals.backend_url +
-                   '/ocs/v2.php/apps/files_sharing/api/v1/shares?format=json&state=all' +
-                   sharedWithMeText
+    params.set('format', 'json')
+    params.set('state', 'all')
+    const apiURL = join(client.globals.backend_url,
+      '/ocs/v2.php/apps/files_sharing/api/v1/shares', `?${params.toString()}`)
+    console.log(apiURL)
```
Comment (individual-it) [inline tests/acceptance/helpers/sharingHelper.js]: debug code?

File: tests/acceptance/helpers/sharingHelper.js [hunk 10]
```diff
@@ -230,8 +238,8 @@ module.exports = {
     for (const element of elementsToDecline) {
       const shareID = element.id
       const headers = httpHelper.createAuthHeader(user)
-      const apiURL = client.globals.backend_url +
-                    '/ocs/v2.php/apps/files_sharing/api/v1/shares/pending/' + shareID + '?format=json'
+      const apiURL = join(client.globals.backend_url, '/ocs/v2.php/apps/files_sharing/api/v1/shares/pending/'
+        , shareID, '?format=json')
```
Comment (individual-it) [inline tests/acceptance/helpers/sharingHelper.js]:
```
      const apiURL = join(
            client.globals.backend_url,
            '/ocs/v2.php/apps/files_sharing/api/v1/shares/pending/',
            shareID,
            '?format=json'
)

File: tests/acceptance/helpers/sharingHelper.js [hunk 11]
```diff
@@ -267,8 +275,8 @@ module.exports = {
     for (const element of elementsToAccept) {
       const shareID = element.id
       const headers = httpHelper.createAuthHeader(user)
-      const apiURL = client.globals.backend_url +
-                      '/ocs/v2.php/apps/files_sharing/api/v1/shares/pending/' + shareID + '?format=json'
+      const apiURL = join(client.globals.backend_url, '/ocs/v2.php/apps/files_sharing/api/v1/shares/pending/',
```
Comment (individual-it) [inline tests/acceptance/helpers/sharingHelper.js]: same as above

File: tests/acceptance/helpers/webdavHelper.js [hunk 12]
```diff
@@ -80,7 +79,7 @@ exports.move = function (userId, fromName, toName) {
 exports.propfind = function (path, userId, properties, folderDepth = 1) {
   const headers = httpHelper.createAuthHeader(userId)
   headers.Depth = folderDepth
-  const davPath = backendHelper.getCurrentBackendUrl() + '/remote.php/dav' + resolve(path)
+  const davPath = join(client.globals.backend_url, '/remote.php/dav', path)
```
Comment (individual-it) [inline tests/acceptance/helpers/webdavHelper.js]: why changing from `getCurrentBackendUrl`?

File: tests/acceptance/helpers/webdavHelper.js [hunk 13]
```diff
@@ -221,15 +221,16 @@ exports.getSkeletonFile = function (filename) {
...
-  return fetch(backendHelper.getCurrentBackendUrl() +
-    '/remote.php/webdav/' + filename, {
-    headers: {
-      'Content-Type': 'text/plain',
-      ...headers
-    },
-    method: 'PUT',
-    body: content
-  })
+  const apiURL = join(client.globals.backend_url, '/remote.php/webdav/', filename)
```
Comment (individual-it) [inline tests/acceptance/helpers/webdavHelper.js]: why changing from `getCurrentBackendUrl`?

File: tests/acceptance/helpers/webdavHelper.js (hunk 12 above)
Comment (jasson99) [reply tests/acceptance/helpers/webdavHelper.js]: fixed

File: tests/acceptance/helpers/webdavHelper.js (hunk 13 above)
Comment (jasson99) [reply tests/acceptance/helpers/webdavHelper.js]: fixed

File: tests/acceptance/helpers/sharingHelper.js (hunk 9 above)
Comment (jasson99) [reply tests/acceptance/helpers/sharingHelper.js]: fixed

File: tests/acceptance/helpers/sharingHelper.js (hunk 10 above)
Comment (jasson99) [reply tests/acceptance/helpers/sharingHelper.js]: fixed

File: tests/acceptance/helpers/sharingHelper.js (hunk 11 above)
Comment (jasson99) [reply tests/acceptance/helpers/sharingHelper.js]: fixed

**Pull request merged**